### PR TITLE
Remove wallet wording from Rewards strings

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/19768

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 1]
```diff
@@ -305,19 +305,19 @@
     }
   },
   "notificationBackupWalletTitle": {
-    "message": "Backup wallet",
+    "message": "Back up wallet",
```
Comment (emerick) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json:308]: Should this be "Back up your Brave Rewards"?

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 2]
```diff
@@ -555,27 +555,27 @@
...
+    "message": "Verify Brave Rewards",
     "description": ""
   },
   "onboardingPanelBitflyerNote": {
     "message": "The bitFlyer wallet is a service managed by bitFlyer, Inc.",
     "description": ""
   },
   "onboardingPanelBitflyerText": {
-    "message": "By verifying your wallet and linking it to a bitFlyer account, you will be able to store and manage the tokens you have earned through Brave Rewards more easily",
+    "message": "By verifying your Brave Rewards and linking it to a bitFlyer account, you will be able to store and manage the tokens you have earned through Brave Rewards more easily",
```
Comment (emerick) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json:578]: I noticed in an earlier string, we renamed "tokens" to "BAT". Not sure if we wanted to do that here as well?

Comment (Miyayes) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json:578]: Good idea, @emerick. We can change that to BAT.

Comment (zenparsing) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json:578]: I've updated this message to use "BAT". There are a few occurrences of "tokens" in the onboarding slides, but we should probably review that as a separate task. Good catch, @emerick!

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json (hunk 1 above)
Comment (zenparsing) [reply components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json:308]: Yep - great catch :)

File: components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json [hunk 3]
```diff
@@ -555,27 +555,27 @@
     "description": ""
   },
   "onboardingPanelVerifyHeader": {
-    "message": "Verify your wallet",
+    "message": "Verify",
     "description": ""
   },
   "onboardingPanelVerifySubtext": {
-    "message": "In order to withdraw the tokens you earn for viewing ads, you’ll need to verify your wallet.",
+    "message": "In order to withdraw the BAT you earn for viewing ads, you’ll need to verify Brave Rewards.",
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/extension/brave_rewards/_locales/en_US/messages.json:562]: I noticed that we mix [right single quotation marks](https://unicode-table.com/en/2019) and [apostrophes](https://unicode-table.com/en/0027) pretty much everywhere. We might want to stick with one of them at some point. 🙂

File: components/resources/brave_components_strings.grd [hunk 4]
```diff
@@ -788,23 +788,23 @@
...
       <message name="IDS_BRAVE_UI_REWARDS_CONTRIBUTE" desc="">Auto-Contribute</message>
       <message name="IDS_BRAVE_UI_REWARDS_CONTRIBUTE_ATTENTION" desc="">Attention</message>
       <message name="IDS_BRAVE_UI_REWARDS_VIEW_QR_CODE_TEXT1" desc="">View QR Code</message>
       <message name="IDS_BRAVE_UI_REWARDS_VIEW_QR_CODE_TEXT2" desc="">for iOS Rewards users transferring BAT.</message>
-      <message name="IDS_BRAVE_UI_REWARDS_RESTORE_TEXT3" desc="">Restoring with a recovery key will replace your current wallet. So make sure you empty or back up your current wallet before restoring.</message>
+      <message name="IDS_BRAVE_UI_REWARDS_RESTORE_TEXT3" desc="">Restoring with a recovery key will replace your current Brave Rewards profile. Make sure you empty or back up your current Brave Rewards before restoring.</message>
       <message name="IDS_BRAVE_UI_REWARDS_RESTORE_TEXT4" desc="">Enter your recovery key or</message>
       <message name="IDS_BRAVE_UI_REWARDS_RESTORE_WARNING" desc="">Backup your wallet before replacing. Or you will lose the fund, [[funds]], in your current wallet.</message>
```
Comment (szilardszaloki) [inline components/resources/brave_components_strings.grd:804]: We might want to change this one, too.

Comment (zenparsing) [reply components/resources/brave_components_strings.grd:804]: I think the new text here matches the replacement text that we came up with? (The "wallet" is replaced with "profile" here.) In any case, I don't think this string is even visible to the user currently - at least not until backup/restore 😄

Comment (szilardszaloki) [reply components/resources/brave_components_strings.grd:804]: I meant this line 🙂:
```
<message name="IDS_BRAVE_UI_REWARDS_RESTORE_WARNING" desc="">Backup your wallet before replacing. Or you will lose the fund, [[funds]], in your current wallet.</message>
```

**Review by szilardszaloki — APPROVED** [review]

File: components/resources/brave_components_strings.grd (hunk 4 above)
Comment (zenparsing) [reply components/resources/brave_components_strings.grd:804]: Ah, gotcha - I think this one also is also not visible right now (although it is referenced in the code somewhere).

**Pull request merged**